File: app/src/main/java/com/example/sy43/database/SubCategory.java
package com.example.sy43.database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(foreignKeys = {@ForeignKey(entity = Category.class, parentColumns = "id",childColumns = "category")})
public class SubCategory {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @ColumnInfo(name = "s_name")
    public String name;

    @ColumnInfo(name = "envelope")
    public double envelope;

    @ColumnInfo(name = "category")
    public int category;

    public String getName() {
        return name;
    }

    public SubCategory(String name, double envelope, int category) {
        this.name = name;
        this.envelope = envelope;
        this.category = category;
    }
}

File: app/src/main/java/com/sy43_p22/piggy_bank/database/entities/SubCategory.kt
package com.sy43_p22.piggy_bank.database.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.CASCADE
import androidx.room.PrimaryKey

@Entity(tableName = "subcategory", foreignKeys = [
    ForeignKey(
        entity = Category::class,
        parentColumns = ["catid"],
        childColumns = ["categoryId"],
        onDelete = CASCADE
    )
])
/**
 * @description Sub-category Class to match up with the Database
 */
data class SubCategory(
    @PrimaryKey(autoGenerate = true)
    /** Sub-category Id */
    val subid: Int = 0,
    /** Sub-category Name */
    val name: String,
    /** Sub-category spending value */
    val spending: Int = 0,
    /** Sub-category saving value */
    val saving: Int = 0,
    @ColumnInfo(index = true)
    /** Parent Category Id */
    val categoryId: Int
)
